qt/scientific_interfaces/Indirect/IndirectFitDataTableModel.cpp

```diff
@@ -0,0 +1,349 @@
+  if (m_fittingData->empty())
+    return;
+  m_fittingData->at(dataIndex.value).setExcludeRegionString(exclude, spectrum);
+}
+
+void IndirectFitDataTableModel::addNewWorkspace(const Mantid::API::MatrixWorkspace_sptr &workspace,
+                                                const FunctionModelSpectra &spectra) {
+  m_fittingData->emplace_back(workspace, spectra);
+}
+
+void IndirectFitDataTableModel::removeWorkspace(TableDatasetIndex index) {
+  if (index < m_fittingData->size()) {
+    m_fittingData->erase(m_fittingData->begin() + index.value);
+  } else {
+    throw std::runtime_error("Attempting to remove non-existent workspace.");
+  }
+}
+
+void IndirectFitDataTableModel::removeDataByIndex(FitDomainIndex fitDomainIndex) {
+  auto subIndices = getSubIndices(fitDomainIndex);
+  auto &spectra = m_fittingData->at(subIndices.first.value).getMutableSpectra();
+  spectra.erase(subIndices.second);
+  // If the spectra list corresponding to a workspace is empty, remove workspace
+  // at this index, else we'll have a workspace persist with no spectra loaded.
+  if (spectra.empty()) {
+    removeWorkspace(subIndices.first.value);
+  }
+}
+
+void IndirectFitDataTableModel::switchToSingleInputMode() {
+  m_fittingData = m_fittingDataSingle.get();
+  m_resolutions = m_resolutionsSingle.get();
+}
+
+void IndirectFitDataTableModel::switchToMultipleInputMode() {
+  m_fittingData = m_fittingDataMultiple.get();
+  m_resolutions = m_resolutionsMultiple.get();
+}
+
+std::vector<double> IndirectFitDataTableModel::getExcludeRegionVector(TableDatasetIndex dataIndex,
+                                                                      WorkspaceIndex index) const {
+  auto fitData = m_fittingData->at(dataIndex.value);
+  return fitData.excludeRegionsVector(index);
+}
+
+Mantid::API::MatrixWorkspace_sptr IndirectFitDataTableModel::getWorkspace(FitDomainIndex index) const {
+  auto subIndices = getSubIndices(index);
+  return getWorkspace(subIndices.first);
+}
+
+std::pair<double, double> IndirectFitDataTableModel::getFittingRange(FitDomainIndex index) const {
+  auto subIndices = getSubIndices(index);
+  return getFittingRange(subIndices.first, subIndices.second);
+}
+
+size_t IndirectFitDataTableModel::getSpectrum(FitDomainIndex index) const {
+  auto subIndices = getSubIndices(index);
+  return subIndices.second.value;
+}
+
+std::vector<double> IndirectFitDataTableModel::getExcludeRegionVector(FitDomainIndex index) const {
+  auto subIndices = getSubIndices(index);
+  return getExcludeRegionVector(subIndices.first, subIndices.second);
+}
+
+std::string IndirectFitDataTableModel::getExcludeRegion(FitDomainIndex index) const {
+  auto subIndices = getSubIndices(index);
+  return getExcludeRegion(subIndices.first, subIndices.second);
+}
+
+void IndirectFitDataTableModel::setExcludeRegion(const std::string &exclude, FitDomainIndex index) {
+  if (m_fittingData->empty())
+    return;
+  auto subIndices = getSubIndices(index);
```

Comment:
We can use `auto const` for a lot of the variables in this function and the above functions